test/e2e/branding_test.go

```diff
@@ -0,0 +1,198 @@
+		if err != nil {
+			t.Fatalf("error: could not apply managed config map %v", err)
+		}
+
+		err = wait.Poll(1*time.Second, pollTimeout, func() (stop bool, err error) {
+			gotBrand := getConsoleBrand(t, client)
+			return (gotBrand == expectedBrand), nil
+		})
+	}
+}
+
+func createConsoleConfigMap(brand operatorsv1.Brand) *v1.ConfigMap {
+	return &v1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ConfigMap",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "console-config",
+			Namespace: "openshift-config-managed",
+		},
+		Data: map[string]string{
+			"console-config.yaml": `kind: ConsoleConfig
+apiVersion: console.openshift.io/v1
+customization:
+  branding: ` + string(brand) + `
+  documentationBaseURL: https://docs.okd.io/4.0/
+`,
+		},
+		BinaryData: nil,
+	}
+}
+
+// Set Brand on the operator config
+func setOperatorConfigBrand(t *testing.T, client *testframework.Clientset, brand operatorsv1.Brand) {
+	operatorConfig, err := client.Consoles().Get(consoleapi.ConfigResourceName, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("could not get operator config, %v", err)
+	}
+	// set it
+	spec := operatorsv1.ConsoleSpec{
+		OperatorSpec: operatorsv1.OperatorSpec{
+			ManagementState: "Managed",
+		},
+		Customization: operatorsv1.ConsoleCustomization{
+			Brand: brand,
+		},
+	}
+	operatorConfig.Spec = spec
+	_, err = client.Consoles().Update(operatorConfig)
+	if err != nil {
+		t.Fatalf("could not update operator config with brand=%v, %v", brand, err)
+	}
+}
+
+// Get the brand from the console-config in the data of the console CM
+func getConsoleBrand(t *testing.T, client *testframework.Clientset) operatorsv1.Brand {
+	cm, err := testframework.GetConsoleConfigMap(client)
+	if err != nil {
+		t.Fatalf("error: %s", err)
+	}
+
+	data := cm.Data["console-config.yaml"]
+	brandingValue := ""
+	temp := strings.Split(data, "\n")
+	for _, item := range temp {
+		if strings.Contains(item, "branding") {
+			brandingValue = strings.Split(strings.TrimSpace(item), ":")[1]
+		}
+	}
+	brand, _ := stringToBrand(strings.TrimSpace(brandingValue))
+	return brand
+}
+
+// Helper function to convert string version of valid brands to their corresponding constant name
+func stringToBrand(brandstr string) (b operatorsv1.Brand, ok bool) {
+	switch found := operatorsv1.Brand(brandstr); found {
+	case
+		operatorsv1.BrandDedicated,
+		operatorsv1.BrandOnline,
+		operatorsv1.BrandOKD,
+		operatorsv1.BrandOCP,
+		operatorsv1.BrandOpenShift,
+		operatorsv1.BrandAzure:
+		return found, true
+	default:
+		fmt.Printf("invalid brand (%s), defaulting to %s \n", brandstr, operatorsv1.BrandOKD)
+		return operatorsv1.BrandOKD, false
+	}
+}
+
+// Helper function that decides whether to update a config map (if it exists) or create a new one
+func updateOrCreateConsoleConfigMap(client *testframework.Clientset, ns string, cm *v1.ConfigMap) (*v1.ConfigMap, error) {
```

Comment:
I think you can drop `ns string` arg and just use the `cm.ObjectMeta.Namespace`.  No need to pass it twice.

Reply:
Can you use `resourceapply.ApplyConfigMap()` instead of this func?  See https://github.com/openshift/console-operator/blob/master/pkg/console/operator/sync_v400.go#L228

Reply:
`ApplyConfigMap` expects a ConfigMapsGetter that is available from the consoleOperator object in the sync routines.  The E2E framework does not have that currently and I am not sure how to add it without a bit of code.  I do agree that `updateOrCreateConsoleConfigMap` function is very similar.

Reply:
So the `clientset` is this:
https://github.com/openshift/console-operator/blob/master/pkg/testframework/clientset.go#L15
can compare to how the operator itself is built up:
https://github.com/openshift/console-operator/blob/master/pkg/console/operator/operator.go#L61

`ConfigMaps` are `core`, so I think you have what you need with `clientset.ConfigMaps()`.